### PR TITLE
Initial support for systemd-boot with archlinux [WIP]

### PR DESCRIPTION
**Pull request by Kamorst**
This should work with a default systemd-boot installation of archlinux.

### CODE REVIEW

File: src/deploy.sh [hunk 1]
```diff
@@ -494,7 +494,7 @@ function prepare_remote_dir()
   fi
 
   target_deploy_path=$(join_path "$target_deploy_path" "$distro.sh")
-  files_to_send="$KW_PLUGINS_DIR/kernel_install/{remote_deploy.sh,utils.sh,$distro.sh,bootloader_utils.sh,grub.sh,rpi_bootloader.sh}"
+  files_to_send="$KW_PLUGINS_DIR/kernel_install/{remote_deploy.sh,utils.sh,$distro.sh,bootloader_utils.sh,grub.sh,rpi_bootloader.sh,systemd_bootloader.sh}"
```
Comment (rodrigosiqueira) [inline src/deploy.sh:497]: `/$KW_PLUGINS_DIR/${KW_PLUGINS_DIR}/`

File: src/plugins/kernel_install/arch.sh [hunk 2]
```diff
@@ -93,8 +93,9 @@ function generate_arch_temporary_root_file_system()
     fi
   fi
 
-  # We do not support initramfs outside grub scope
-  [[ "$bootloader_type" != 'GRUB' ]] && return
+  # We do not support initramfs outside grub and systemd-boot scope
+  [[ "$bootloader_type" != 'GRUB' && "$bootloader_type" != 'SYSTEMD_BOOT' ]] && return
```
Comment (rodrigosiqueira) [inline src/plugins/kernel_install/arch.sh:97]: Wait, what? Systemd does use initramfs? Does it use something else?

File: src/plugins/kernel_install/systemd_bootloader.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
+function run_bootloader_update()
```
Comment (rodrigosiqueira) [inline src/plugins/kernel_install/systemd_bootloader.sh:1]: Don't forget to add documentation to this file.

File: src/plugins/kernel_install/utils.sh [hunk 4]
```diff
@@ -361,12 +361,18 @@ function update_bootloader()
 
   case "${deploy_data['bootloader']}" in
     GRUB)
+      echo 'grub identificado'
```
Comment (rodrigosiqueira) [inline src/plugins/kernel_install/utils.sh:364]: Don't forget to remove this.

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]
Very nice PR! You are in the right direction.

File: src/plugins/kernel_install/arch.sh (hunk 2 above)
Comment (Kamorst) [reply src/plugins/kernel_install/arch.sh:97]: The systemd-boot (and grub and the efi firmware) could boot a unified kernel image which embeds the initramfs and the kernel in the same efi executable, but it is not mandatory and would need more changes to the kw deploy system to generate those file.

File: src/plugins/kernel_install/systemd_bootloader.sh [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  local flag="$1"
+  local target="$2"
+  local name="$3"
+  local sudo_cmd
+  local loader_conf="${SYSTEMD_BOOT_ENTRIES_DIR}/kw_${name}_linux.conf"
+
+  #if [[ ! -e "/boot/vmlinuz-${name}" && -e "$loader_conf" ]]; then
+  #  echo removing boot entry
+  #  remove_systemd_boot_entry "$flag" "$name"
+  #fi
```
Comment (rodrigosiqueira) [inline src/plugins/kernel_install/systemd_bootloader.sh:14]: Remember to drop this comment.

File: src/plugins/kernel_install/systemd_bootloader.sh [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+  if [[ -e "/boot/vmlinuz-${name}" ]]; then
+    add_systemd_boot_entry "$flag" "$target" "$name"
+  fi
+}
+
+function add_systemd_boot_entry()
```
Comment (rodrigosiqueira) [inline src/plugins/kernel_install/systemd_bootloader.sh:21]: Don't forget to add documentation to this file.

File: src/plugins/kernel_install/systemd_bootloader.sh [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+{
+  local flag="$1"
+  local target="$2"
+  local name="$3"
+  local loader_conf="${SYSTEMD_BOOT_ENTRIES_DIR}/kw_${name}_linux.conf"
+  local kernel_parameters
+
+  [[ "$target" == 'local' ]] && sudo_cmd='sudo'
```
Comment (rodrigosiqueira) [inline src/plugins/kernel_install/systemd_bootloader.sh:29]: I think you forgot to use `sudo_cmd` for local deploy. We don't need `sudo` for remote deploy.

File: src/plugins/kernel_install/systemd_bootloader.sh [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+  } | sudo tee "$loader_conf" > /dev/null
+}
+
+function remove_systemd_boot_entry()
+{
+  local flag="$1"
+  local name="$2"
+  local loader_conf="${SYSTEMD_BOOT_ENTRIES_DIR}/kw_${name}_linux.conf"
+
+  sudo rm "$loader_conf"
```
Comment (rodrigosiqueira) [inline src/plugins/kernel_install/systemd_bootloader.sh:51]: You probably want:
```
  [[ "$target" == 'local' ]] && sudo_cmd='sudo'
```
and use `cmd_manager` here.

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]

**Comment by rodrigosiqueira** [comment]
Any progress here?

**Comment by rodrigosiqueira** [comment]
@Kamorst ?

**Pull request closed without merging**